yorkie/backend/sync/etcd/pubsub.go

```diff
@@ -41,11 +48,73 @@ func (c *Client) Publish(
 	publisherID *time.ActorID,
 	topic string,
 	event sync.DocEvent,
+) {
+	c.PublishToLocal(publisherID, topic, event)
+
+	c.memberMapMu.RLock()
+	defer c.memberMapMu.RUnlock()
+
+	for _, member := range c.Members() {
+		memberAddr := member.RPCAddr
+		if memberAddr != c.pubSub.AgentInfo.RPCAddr {
+			c.addBroadcastClient(memberAddr)
+		}
+
+		if memberAddr != c.pubSub.AgentInfo.RPCAddr {
+			c.publishToMember(
+				memberAddr,
+				publisherID,
+				topic,
+				event,
+			)
+		}
+	}
+}
+
+// PublishToLocal publishes the given event to the given Topic.
+func (c *Client) PublishToLocal(
+	publisherID *time.ActorID,
+	topic string,
+	event sync.DocEvent,
 ) {
 	c.pubSub.Publish(publisherID, topic, event)
+}
+
+// addBroadcastClient activates the broadcast grpc server and creates a client.
+func (c *Client) addBroadcastClient(addr string) {
+	if _, ok := c.broadcastClientMap[addr]; !ok {
+		conn, err := grpc.Dial(addr, grpc.WithInsecure())
+		if err != nil {
+			log.Logger.Error(err)
+		}
+
+		c.broadcastClientMap[addr] = &broadcastClientInfo{
+			client: api.NewBroadcastClient(conn),
+			conn:   conn,
+		}
+	}
+}
+
+// publishToMember publishes events to other agents.
+func (c *Client) publishToMember(
+	memberAddr string,
+	publisherID *time.ActorID,
+	topic string,
+	event sync.DocEvent,
+) {
+	bcm := c.broadcastClientMap[memberAddr]
+
+	docEvent, err := converter.ToDocEvent(event)
+	if err != nil {
+		log.Logger.Error(err)
+		return
```

Comment:
Is there any reason we broke the error chain here?

Reply:
I didn't think it was necessary to return an error from the API where a particular Document event occurred just because it couldn't broadcast to another agent.
So I left an error log at the time of occurrence. This is because we thought that document sync could be synced through other sync events.

I have been thinking about how to alert when a problem that can occur within the method (such as a convert failure or an abnormal problem in the target agent) occurs continuously, but I was not sure, so I tried to suggest it later.

Can you please let me know if there are more things I should consider?

Reply:
When we try to convert an event type that is not supported, an error occurs, and there is no need to loop further in the caller. I wish we could handle errors explicitly.